plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/gitlabMergeRequest.ts

```diff
@@ -0,0 +1,197 @@
+import { ScmIntegrationRegistry } from '@backstage/integration';
+import { InputError } from '@backstage/errors';
+import { parseRepoUrl } from './util';
+
+export type GitlabMergeRequestActionInput = {
+  projectid: string;
+  repoUrl: string;
+  title: string;
+  description: string;
+  destinationBranch: string;
+  targetPath: string;
+};
+
+export const createPublishGitlabMergeRequestAction = (options: {
+  integrations: ScmIntegrationRegistry;
+}) => {
+  const { integrations } = options;
+
+  return createTemplateAction<GitlabMergeRequestActionInput>({
+    id: 'publish:gitlab-merge-request',
+    schema: {
+      input: {
+        required: ['projectid', 'repoUrl', 'targetPath'],
+        type: 'object',
+        properties: {
+          repoUrl: {
+            type: 'string',
+            title: 'Repository Location',
+            description: `Accepts the format 'gitlab.com/group_name/project_name' where 'project_name' is the repository name and 'group_name' is a group or username`,
+          },
+          projectid: {
+            type: 'string',
+            title: 'projectid',
+            description: 'Project ID of the Gitlab Project',
+          },
+          title: {
+            type: 'string',
+            title: 'Merge Request Name',
+            description: 'The name for the merge request',
+          },
+          description: {
+            type: 'string',
+            title: 'Merge Request Description',
+            description: 'The description of the merge request',
+          },
+          destinationBranch: {
+            type: 'string',
+            title: 'Destination Branch name',
+            description: 'The description of the merge request',
+          },
+          targetPath: {
+            type: 'string',
+            title: 'Repository Subdirectory',
+            description: 'Subdirectory of repository to apply changes to',
+          },
+        },
+      },
+      output: {
+        type: 'object',
+        properties: {
+          projectid: {
+            title: 'Gitlab Project id',
+            type: 'string',
+          },
+          mergeRequestURL: {
+            title: 'MergeRequest(MR) URL',
+            type: 'string',
+            description: 'Link to the merge request in GitLab',
+          },
+        },
+      },
+    },
+    async handler(ctx) {
+      const repoUrl = ctx.input.repoUrl;
+      const { host } = parseRepoUrl(repoUrl, integrations);
+      const integrationConfig = integrations.gitlab.byHost(host);
+
+      const actions: CommitAction[] = [];
+      const formatedTimestamp = () => {
+        const d = new Date();
+        const date = d.toISOString().split('T')[0];
+        const time = d.toTimeString().split(' ')[0].replace(/:/g, '_');
+        return `${date}_${time}`;
+      };
+      const destinationBranch = ctx.input.destinationBranch
+        ? ctx.input.destinationBranch + formatedTimestamp()
```

Comment:
I would somewhat expect the branch to be passed through as is if you set one tbh. There's opportunity for templating the name of it from the outside too if needed.

Reply:
@Rugvip do you mean user should be able to choose the destination branch where the MR need to be created?. It is already template in that way.